tools/nstar/src/main.rs

```diff
@@ -342,6 +355,61 @@ async fn main() -> Result<()> {
                 process::exit(0);
             }
         }
+        // Start process inside running container and attach a pseudo-terminal to it
+        Subcommand::Exec { name, version, cmd } => {
+            // Connect
+            let mut client = Client::new(io, None, opt.timeout)
+                .await
+                .context("Failed to connect")?;
+
+            let version = or_version(&name, version, &mut client).await?;
+            let cmd: Result<Vec<_>> = cmd
+                .into_iter()
+                .map(|c| NonNullString::try_from(c).context("Invalid arg"))
+                .collect();
+
+            // TODO Avoid the nix dependency by opening the PTY in the runtime and sending the
+            // master FD through the client socket.
```

Comment:
This doesn't work if nstar is connected via TCP. Right?

Reply:
Good point, in any case `exec` only works if executed locally :sweat:

Reply:
Why should it? If the RT takes care for bridging between the connection and the pty, TCP should be also possible.
Only if we create the  pty in nstar this would be a problem.

Reply:
moved the pty opening to the runtime and use an async task to forward the input/output from the client connection to the container.